File: data-structures/stack/implementation.java
public class Stack {
    private int[] arr;
    private int capacity;
    private int top;

    public Stack(int size) {
        capacity = size;
        arr = new int[capacity];
        top = -1;
    }

    public void push(int val) {
        if (top == capacity - 1) return;
        arr[++top] = val;
    }

    public void pop() {
        if (top == -1) return;
        top--;
    }

    public int top() {
        if (top == -1) return -1;
        return arr[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }
}

File: data-structures/stack/implementation.cpp
#include <iostream>
using namespace std;

class Stack {
private:
    int* arr;
    int capacity;
    int topIndex;

public:
    Stack(int size = 100) {
        capacity = size;
        arr = new int[capacity];
        topIndex = -1;
    }

    ~Stack() {
        delete[] arr;
    }

    void push(int val) {
        if (topIndex == capacity - 1) return;
        arr[++topIndex] = val;
    }

    void pop() {
        if (topIndex == -1) return;
        topIndex--;
    }

    int top() {
        if (topIndex == -1) return -1;
        return arr[topIndex];
    }

    bool isEmpty() {
        return topIndex == -1;
    }
};
